### PR TITLE
fix: remove unnecessary conversion and several scope lint errors

### PR DESCRIPTION
**Pull request by draveness**
/kind cleanup
/priority backlog
/sig scheduling

**What this PR does / why we need it**:

remove unnecessary conversion and several scope lint errors

```release-note
NONE
```

### CODE REVIEW

File: pkg/scheduler/eventhandlers_test.go [hunk 1]
```diff
@@ -104,7 +104,9 @@ func TestSkipPodUpdate(t *testing.T) {
 			expected: false,
 		},
 	}
-	for _, test := range table {
+	for i := range table {
+		test := table[i]
```
Comment (miguelbernadi) [inline pkg/scheduler/eventhandlers_test.go]: Why this change? It's functionally identical with one lookup and one allocation more than before.

File: pkg/scheduler/scheduler_test.go [hunk 2]
```diff
@@ -275,7 +275,9 @@ func TestScheduler(t *testing.T) {
 	informerFactory.Start(stop)
 	informerFactory.WaitForCacheSync(stop)
 
-	for _, item := range table {
+	for i := range table {
+		item := table[i]
+
```
Comment (miguelbernadi) [inline pkg/scheduler/scheduler_test.go]: Same as above. https://github.com/kubernetes/kubernetes/pull/78680/files#r290218168

File: pkg/scheduler/eventhandlers_test.go (hunk 1 above)
Comment (draveness) [reply pkg/scheduler/eventhandlers_test.go]: the current violates scopelint check and may cause data race problem as https://github.com/kubernetes/kubernetes/pull/78563#discussion_r289311252

Comment (draveness) [reply pkg/scheduler/eventhandlers_test.go]: FYI: https://github.com/kyoh86/scopelint#whats-this

**Comment by poothia** [comment]
lgtm

**Comment by wgliang** [comment]
There should be a lot of such errors. Can you fix all of them in this PR?

**Comment by draveness** [comment]
> There should be a lot of such errors. Can you fix all of them in this PR?

I ran `scopelint` against files under `./pkg/...` and found out there are 3498 lint errors. I'm afraid that is impossible to fix within this PR. I'll open an issue to track this and ask suggestions from other members.

**Comment by draveness** [comment]
> There should be a lot of such errors. Can you fix all of them in this PR?

FYI https://github.com/kubernetes/kubernetes/issues/78689

File: pkg/scheduler/eventhandlers_test.go (hunk 1 above)
Comment (miguelbernadi) [reply pkg/scheduler/eventhandlers_test.go]: Thanks for the explanation! I wasn't aware of this.
